parity-crypto/src/publickey/ecdsa_signature.rs

```diff
@@ -247,8 +247,9 @@ pub fn verify_address(address: &Address, signature: &Signature, message: &Messag
 /// Recovers the public key from the signature for the message
 pub fn recover(signature: &Signature, message: &Message) -> Result<Public, Error> {
 	let context = &SECP256K1;
+	let secp_msg = ZeroesAllowedMessage(message.clone());
```

Comment:
`Message` (which is an alias to `H256` here) is a copy-type, so a deref should be enough to copy it
```suggestion
	let secp_msg = ZeroesAllowedMessage(*message);
```

Reply:
Nice. It's still a copy though right?